src/modules/fw_pos_control/FixedwingPositionControl.cpp

```diff
@@ -1079,7 +1079,13 @@ FixedwingPositionControl::control_auto_position(const float control_interval, co
 
 	if (_position_setpoint_previous_valid && pos_sp_prev.type != position_setpoint_s::SETPOINT_TYPE_TAKEOFF) {
 		Vector2f prev_wp_local = _global_local_proj_ref.project(pos_sp_prev.lat, pos_sp_prev.lon);
-		navigateWaypoints(prev_wp_local, curr_wp_local, curr_pos_local, ground_speed, _wind_vel);
+
+		if (isDoingBackTransition()) {
+			navigateLine(prev_wp_local, curr_wp_local, curr_pos_local, ground_speed, _wind_vel);
+
+		} else {
+			navigateWaypoints(prev_wp_local, curr_wp_local, curr_pos_local, ground_speed, _wind_vel);
+		}
```

Comment:
@tstastny What we can do shorterm is maybe to move this code into a `control_backtransition` method? In a follow up PR to this I am actually trying to make navigator set the setpoint type to "LAND" which should give the position controller and idea that we are currently doing a vtol landing.
Then we don't need to mix the logic with the standard waypoint following logic. What do you think?

Reply:
> control_backtransition

Could be one way - at least to start splitting out the functionalities.. and then once we can do a larger module split.. the methods are already delineated and ready to migrate

> In a follow up PR to this I am actually trying to make navigator set the setpoint type to "LAND" which should give the position controller and idea that we are currently doing a vtol landing.

I don't fully get what you mean -- but maybe this we could chat offline

> Then we don't need to mix the logic with the standard waypoint following logic. What do you think?

Would be the idea

Reply:
Currently, the position controller has no clue that he's steering towards a transition waypoint. This is because navigator just publishes a standard position waypoint. If navigator does not do that but just keeps the original intent (which is to land), then the position controller can call the appropriate control functions.

And yes, I agree that the logic should then at some point be pulled into a mode. But as you said, it's clearly easier to do that refactor if the code is already split up nicely in the controller now.

Reply:
> Currently, the position controller has no clue that he's steering towards a transition waypoint. This is because navigator just publishes a standard position waypoint. If navigator does not do that but just keeps the original intent (which is to land), then the position controller can call the appropriate control functions.

Ook i understand you now. Yes that would be a nice start!